flixkit.go

```diff
@@ -122,36 +91,28 @@ func (s *flixServiceImpl) GetFlixByID(ctx context.Context, templateID string) (*
 	return parsedTemplate, nil
 }
 
-func (t *FlowInteractionTemplate) IsScript() bool {
-	return t.Data.Type == "script"
-}
-
-func (t *FlowInteractionTemplate) IsTransaction() bool {
-	return t.Data.Type == "transaction"
-}
-
-func (t *FlowInteractionTemplate) GetAndReplaceCadenceImports(networkName string) (string, error) {
-	cadence := t.Data.Cadence
 
-	for dependencyAddress, contracts := range t.Data.Dependencies {
-		for contractName, networks := range contracts {
-			network, ok := networks[networkName]
-			if !ok {
-				return "", fmt.Errorf("network %s not found for contract %s", networkName, contractName)
-			}
+func (s *flixServiceImpl) GenFlixBinding(ctx context.Context, templateLocation string, lang string, isLocal bool) (string, error) {
+	var template string
+	var err error
+	if isLocal {
+		template, err = FetchFlixWithContextFromFile(s.config.FileReader, ctx, templateLocation)
```

Comment:
in idiomatic go ctx is always the first parameter sent in to a function if you send it in.